### PR TITLE
Modifications to `doslearn` integration

### PR DESCRIPTION
**Pull request by jwa7**

### CODE REVIEW

File: rholearn/doslearn/train.py [hunk 1]
```diff
@@ -77,42 +80,60 @@ def train():
...
+        descriptor = torch.load(join(ml_options["ML_DIR"], "descriptors.pt"))
+    else:
+        # Compute descriptor
+        io.log(log_path, "Computing SOAP power spectrum descriptors")
+        descriptor = model.compute_descriptor(frames).to(dtype=ml_options["TRAIN"]["dtype"], device=ml_options["TRAIN"]["device"])
+        descriptor = [
+            mts.slice(descriptor, "samples", mts.Labels(["system"], torch.tensor([A]).reshape(-1, 1)))
+            for A in frame_idxs
+        ]
+        torch.save(descriptor, join(ml_options["ML_DIR"], "descriptors.pt")) # Is this the best way to save it?
```
Comment (jwa7) [inline rholearn/doslearn/train.py]:
```suggestion
    if (yaml option load descriptors from path)
        descriptors = [load_tensormap(os.path.join(path, frame_idx)) for frame_idx in frame_idxs]
    else
        descriptors = model.compute_descriptor(frames, frame_idxs)
        descriptors = [mts.slice(...) ...]
        if save_descriptors:
            [mts.save(d, "path/to/checkpoint/frame_idx/descriptor.npz") for d in descriptors]
```

File: rholearn/doslearn/train.py [hunk 2]
```diff
@@ -77,42 +80,60 @@ def train():
...
+    # Check if splines have been computed
+    splines_saved = os.path.exists(join(ml_options["ML_DIR"], "splines.pt"))
+    if descriptor_saved:
```
Comment (jwa7) [inline rholearn/doslearn/train.py]:
```suggestion
    if splines_precomputed:
```

File: rholearn/doslearn/train.py [hunk 3]
```diff
@@ -77,42 +80,60 @@ def train():
...
-        parser.spline_eigenenergies(
-            frame, frame_idx, energies, sigma=ml_options["DOS"]["sigma"], min_energy=ml_options["DOS"]["min_energy"], max_energy=ml_options["DOS"]["max_energy"], interval=ml_options["DOS"]["interval"]
-        )
-        for frame, frame_idx, energies in zip(frames, frame_idxs, eigvals)
-    ]
+        io.log(log_path, "Parsing raw eigenvalues from FHI-aims output directories")
+        eigvals = [
+            parser.parse_eigenvalues(ml_options["SCF_DIR"](A)) for A in frame_idxs
+        ]
+        eigvals = [i[:2] for i in eigvals] # For debugging, remove :2 during final run
```
Comment (jwa7) [inline rholearn/doslearn/train.py]: To remove
```suggestion
        eigvals = [i[:2] for i in eigvals] # For debugging, remove :2 during final run
```

File: example/doslearn-aims-tutorial/part-2-ml/ml-options.yaml [hunk 4]
```diff
@@ -4,7 +4,8 @@ N_VAL: 2
 N_TEST: 2
 
 # Other user settings here:
-# ...
+ML_DIR : "./"
+CHKPT_DIR: "./checkpoints/"
```
Comment (jwa7) [inline example/doslearn-aims-tutorial/part-2-ml/ml-options.yaml]: not needed
```suggestion
```

File: rholearn/doslearn/train.py (hunk 1 above)
Comment (HowWeiBin) [reply rholearn/doslearn/train.py]: Removed saving for now, will provide saving support in a future release

File: rholearn/doslearn/train.py (hunk 2 above)
Comment (HowWeiBin) [reply rholearn/doslearn/train.py]: Renamed

File: rholearn/doslearn/train.py (hunk 3 above)
Comment (HowWeiBin) [reply rholearn/doslearn/train.py]: Fixed

File: example/doslearn-aims-tutorial/part-2-ml/ml-options.yaml (hunk 4 above)
Comment (HowWeiBin) [reply example/doslearn-aims-tutorial/part-2-ml/ml-options.yaml]: Fixed

**Comment by HowWeiBin** [comment]
Resolved issues, in latest commit

**Pull request merged**